src/eval_context.rs

```diff
@@ -196,7 +194,7 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
             Struct(_)    => unimplemented!(),
             Tuple(_)     => unimplemented!(),
-            Function(_)  => unimplemented!(),
+            Function(_, _)  => unimplemented!(),
```

Comment:
Tempted to make this the way to refer to a `TyFnDef` (now that it has `Substs`), what do you think?